js/controllers.js

```diff
@@ -814,6 +814,7 @@ define(['ajv'], function (Ajv) {
       }
       $rootScope.$emit(self.selectionChangedEvent, {
         item:self.selectedItem,
+        segmentNumber:$scope.segmentNumber,
```

Comment:
There is a segmentNumber comparison so that the current segment (displayed tab segment number) matches the segmentNumber for the selector so that not accidentally another segments attribute get changed (emitted)